src/ophyd_async/core/_providers.py

```diff
@@ -17,43 +20,155 @@ class DirectoryInfo:
 
     :param root: Path of a root directory, relevant only for the file writer
     :param resource_dir: Directory into which files should be written, relative to root
-    :param prefix: Optional filename prefix to add to all files
-    :param suffix: Optional filename suffix to add to all files
+    :param filename: Base filename to use generated by FilenameProvider
+    :param create_dir_depth: Optional depth of dirs to create if they do not exist
     """
 
     root: Path
     resource_dir: Path
-    prefix: Optional[str] = ""
-    suffix: Optional[str] = ""
+    filename: str
+    create_dir_depth: Optional[int] = -3
```

Comment:
Does a negative number have specific meaning? Must document: negatives, 0, positives, None. Is None the same as 0, if so can be non-Optional but default to 0.

Reply:
I know that Mark Rivers' AD uses a negative number here. At some point, @jwlodek, I think we brainstormed how best to handle this and agreed that it might be best to accept positive or negative here and use `abs` to pass the right thing to EPICS.

I would vote for making this not `Optional`.

Reply:
You can see my comment on the PR in the PandA IOC about how this number works: https://github.com/PandABlocks/PandABlocks-ioc/pull/102#issuecomment-2117552752

Reply:
This is now `create_dir_depth: int = 0` which looks correct with the ad fileplugin docs.